IPython/html/static/notebook/js/kernelselector.js

```diff
@@ -169,7 +169,8 @@ define([
         var that = this;
         if (typeof selected === 'string') {
             selected = {
-                name: selected
+                name: selected,
+                language: selected
```

Comment:
Answered below. When given as a single string, this should only be the kernelspec name. If you want to specify different information, use a dict.